### DISCUSSION TITLE
linux log formats

### DISCUSSION THREAD

**Discussion by yadheen** [SigmaHQ/sigma#4440] [Q&A] [answered]
Is there any format or template for the linux profiles(other than service mentioned profiles) ? Or any references of the site to look up....

**Comment by nasbench** [comment]
Can you elaborate on what you mean by linux profiles?

**Reply by yadheen** [reply]
I meant the sigma rules like "rules/linux/process_creation/proc_creation_lnx_clear_syslog.yml". Other sigma rules like "rules/linux/auditd/lnx_auditd_audio_capture.yml" have services mentioned in the logsource, I can find the log formats for auditd, auth,... services . But i didnt get any formats for process_creation, network_connections, file_events(category in logsources),... Or is there any other way to check the log formats of these categories...?

**Reply by nasbench** [reply, marked as answer]
Ah you're talking about the logsource field. Basically you can check the taxonomy file which contains all the currently supported logsources https://github.com/SigmaHQ/sigma-specification/blob/main/Taxonomy_specification.md

The difference between category, service and product can be found here https://github.com/SigmaHQ/sigma-specification/blob/main/Sigma_specification.md#log-source

Basically the the builtin folder contains rules that are from logs that are generated by builtin services. The other such as "process_creation" / "file_event" / "network_connection" are generic mappings based on Sysmon for linux or any similar EDR like tool that can generate those events.

Hope this clear things up for you.

**Reply by yadheen** [reply]
oh! thanks for the clearance... thill will help me. But why sysmon isn't mentioned in logsource

**Reply by nasbench** [reply]
Sysmon doesn't need to be mentioned as its a special case. Unless you're writing rules specifically for it then you select the "service: sysmon". But in general the category includes a much wider range of services. You could map "process_creation" to Sysmon logs, Linux EDR logs...etc.

**Reply by yadheen** [reply]
yeah.. thanks man 👍

**Comment by yadheen** [comment]
I got another thing to discuss... "rules/linux/auditd/lnx_auditd_omigod_scx_runasprovider_executeshellcommand.yml" this rules contains both 'cwd' and other fields under selection identifier, which are from separate log line I think.  shouldn't this be written like correlated events with separate identifiers. 
https://user-images.githubusercontent.com/9653181/133756156-4fb9c2b1-aa65-4380-957b-72170de36fc4.png

**Reply by nasbench** [reply]
Yes. In its current form it wouldn't work except if someone actually correlated the log. It needs a fix and a disclaimer (that it'll get).
Thanks for reporting it
